src/mgr/ServiceMap.cc

```diff
@@ -75,20 +77,101 @@ std::string ServiceMap::Service::get_summary() const
   if (daemons.empty()) {
     return "no daemons active";
   }
+
   std::ostringstream ss;
-  ss << daemons.size() << (daemons.size() > 1 ? " daemons" : " daemon")
-     << " active";
 
-  if (!daemons.empty()) {
-    ss << " (";
+  // The format is "type/prefix:daemon_name".
+  // "type": will be used to replace the default "daemon(s)"
+  // showed in `ceph -s`. If absent, the "daemon" will used.
+  // "prefix": if present the active members will be classified
+  // by the "prefix" instead of "daemon_name".
+  //
+  // For exmaple for iscsi gateways, it will be something likes:
+  // "portal/hostX:rbd/imageX"
```

Comment:
Nit: I wonder if it would be cleaner to just inject this prefix into the daemon metadata? That would ensure that we don't break backwards compatibility for dashboard + ceph-iscsi (which can be upgraded independently) and would also allow the "tcmu-runner" daemon to be listed as "iscsi" or similar via a second key.

Reply:
Yeah, this is feasible.